src/org/labkey/snd/SNDManager.java

```diff
@@ -4174,24 +4171,25 @@ private Map<Integer, Map<Integer, SuperPackage>> getNextLevelEventDataSuperPkgs(
                                     )
                             )
                     );
-            eventDataSuperPkgIds = nextLevelEventDataSuperPkgs.containsKey(eventData.getEventId())
-                    ? nextLevelEventDataSuperPkgs.get(eventData.getEventId()).values().stream().map(SuperPackage::getSuperPkgId).toList()
-                    : new ArrayList<>();
         }
-        else
-        {
-            nextLevelEventDataSuperPkgs = new HashMap<>();
+
+        if (!nextLevelEventDataSuperPkgs.containsKey(eventData.getEventId())) {
             nextLevelEventDataSuperPkgs.put(eventData.getEventId(), new HashMap<>());
-            eventDataSuperPkgIds = new ArrayList<>();
         }
 
-        AtomicInteger emptyEventDataId = new AtomicInteger(0); 
 
-        if (includeEmptySubPackages && nextLevelEventDataSuperPkgs.containsKey(eventData.getEventId())) {
+        if (includeEmptySubPackages) {
+
+            List<Integer> eventDataSuperPkgIds = !nextLevelEventDataSuperPkgs.get(eventData.getEventId()).isEmpty()
+                    ? nextLevelEventDataSuperPkgs.get(eventData.getEventId()).values().stream().map(SuperPackage::getSuperPkgId).toList()
+                    : new ArrayList<>();
+
+            AtomicInteger emptyEventDataId = new AtomicInteger(0);
```

Comment:
Can i please know if we are using any multithreads in here as i do see AtomicInteger? i checked in the sndManager code and i don't see any threads being used in this file.

Reply:
No, but in order to iterate an Integer object within a stream it needs to be Atomic